core/src/bundle_stage.rs

```diff
@@ -672,35 +650,54 @@ impl BundleStage {
             if bundles.len() == BUNDLE_WINDOW_SIZE.get() {
                 // unwrwap safe here because of the length check
                 let bundle = bundles.pop_front().unwrap();
-                if let Some(output) = Self::process_bundle(
-                    bank,
-                    bundle,
-                    bundle_storage,
-                    bundle_account_locker,
-                    consumer,
-                    consume_worker_metrics,
-                ) {
-                    consume_worker_metrics.update_for_consume(&output);
-                    consume_worker_metrics.set_has_data(true);
-                    bundle_stage_metrics.increment_bundles_processed(1);
+
+                let result = Self::process_bundle(bank, &bundle, consumer, consume_worker_metrics);
```

Comment:
manage bundle locks and bundle storage better. bundles should always be destroyed on ok or not retryable and retried on retryable error.